src/bp/ui-studio/src/web/components/Layout/StatusBar/ConfigStatus.tsx

```diff
@@ -0,0 +1,100 @@
+import { Button, Classes, Colors, Dialog, Icon, Intent } from '@blueprintjs/core'
+import axios from 'axios'
+import React, { Fragment, useEffect, useState } from 'react'
+import { toastFailure } from '~/components/Shared/Utils'
+import EventBus from '~/util/EventBus'
+
+import ActionItem from './ActionItem'
+import style from './StatusBar.styl'
+
+const adminUrl = `${window['ROOT_PATH']}${window['API_PATH']}/admin/server/`
+
+const ConfigStatus = () => {
+  const [isDifferent, setDifferent] = useState(false)
+  const [isRestarting, setRestart] = useState(false)
+  const [isOpen, setOpen] = useState(false)
+
+  useEffect(() => {
+    // tslint:disable-next-line: no-floating-promises
+    fetchHash()
+
+    const configUpdated = event => setDifferent(event.initialHash !== event.newHash)
+
+    EventBus.default.on('config.updated', configUpdated)
+    return () => EventBus.default.off('config.updated', configUpdated)
+  }, [])
+
+  useEffect(() => {
+    if (!isRestarting) {
+      return
+    }
+
+    const interval = setInterval(async () => {
+      try {
+        await axios.get('/status', { timeout: 500 })
+        window.location.reload()
+      } catch (err) {} // silent catch intended
+    }, 1000)
+    return () => clearInterval(interval)
+  }, [isRestarting])
+
+  const fetchHash = async () => {
+    try {
+      const { data } = await axios.get(`${adminUrl}/configHash`)
+      if (data.initialHash && data.currentHash && data.initialHash !== data.currentHash) {
+        setDifferent(true)
+      }
+    } catch (err) {
+      toastFailure(err.message)
+    }
+  }
+
+  const restartServer = async () => {
```

Comment:
I suggest either a window.confirm or a toggle or somekind of warning just to make sure the user doesn't hit this by mistake. 

\+ we might want to add a permission check on this as IMO an external bot developer shouldn't be able to neither see this nor to reboot server

Reply:
Only super admins see the ConfigStatus component in the status bar. When you click on the icon, you see a dialog explaining the required reboot, and there is a button to do the restart. The dialog is the confirm window